File: example-java/src/main/java/io/github/cmh1448/examplejava/configuration/SecurityConfiguration.java
package io.github.cmh1448.examplejava.configuration;

import io.github.cmh1448.autojwt.configurer.JwtConfigurerFactory;
import io.github.cmh1448.autojwt.service.UserLoadService;
import io.github.cmh1448.examplejava.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.web.SecurityFilterChain;

@Configuration
@EnableWebSecurity
@RequiredArgsConstructor
public class SecurityConfiguration {
    private final JwtConfigurerFactory jwtConfigurerFactory;
    private final UserService userService;

    @Bean
    public PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }

    @Bean
    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
        http.httpBasic(AbstractHttpConfigurer::disable);

        jwtConfigurerFactory.createJwtAutoConfigurer(userService)
                .pathConfigure(it -> {
                    it.includeAll();
                    it.excludePattern("/register");
                    it.excludePattern("/login");
                })
                .configure(http);

        return http.build();
    }
}

File: example/src/main/kotlin/io/github/cmh1448/autojwt/example/configuration/SecurityConfiguration.kt
package io.github.cmh1448.autojwt.example.configuration

import io.github.cmh1448.autojwt.configurer.JwtConfigurerFactory
import io.github.cmh1448.autojwt.service.UserLoadService
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.web.SecurityFilterChain

@Configuration
@EnableWebSecurity
class SecurityConfiguration(
    private val jwtAutoConfigurerFactory: JwtConfigurerFactory,
    private val userLoadService: UserLoadService
)  {
    @Bean
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
        val jwtAutoConfigurer = jwtAutoConfigurerFactory.createJwtAutoConfigurer(userLoadService)

        jwtAutoConfigurer
            .pathConfigure {
                it.includeAll()
                it.excludePattern("/register")
                it.excludePattern("/login")
            }
            .configure(http)

        return http.build()
    }

    @Bean
    fun passwordEncoder() = BCryptPasswordEncoder()

}